app/src/models/location.js

```diff
@@ -0,0 +1,121 @@
+module.exports = (sequelize, DataTypes) => {
+  const Location = sequelize.define('Location', {
+    locationId: {
+      allowNull: false,
+      defaultValue: DataTypes.UUIDV4,
+      primaryKey: true,
+      type: DataTypes.UUID,
+      validate: {
+        isUUID: 4
+      }
+    },
+    startDate: {
+      allowNull: false,
+      type: DataTypes.DATE
+    },
+    endDate: {
+      allowNull: false,
+      type: DataTypes.DATE
```

Comment:
I've asked about optionality of these.
Enforcing entry on the front end for now too.